mac/hid.c

```diff
@@ -1149,6 +1149,11 @@ static int get_report(hid_device *dev, IOHIDReportType type, unsigned char *data
 
 	register_device_error(dev, NULL);
 
+	if (!data || (length == 0)) {
+		register_device_error(dev, "Zero buffer/length");
```

Comment:
same here

Reply:
there is even bigger issue here, that I missed earlier - we access `data` before we check for it/its length

Reply:
fixed